UI/Components/TheoryComparisonGeneratorSettings.cs

```diff
@@ -345,33 +345,43 @@ private void _unlockFile()
                 File.Delete(TheoryTimesFilePathLock);
             }
         }
-
-        private void _theoryComparisonGeneratorSettings_Leave(object sender, EventArgs e)
+        private void _LayoutSettingsClosed(object sender, EventArgs e)
         {
             var shouldSave = false;
 
-            if (HasChanged)
-            {
-                var result = MessageBox.Show(
-                    "Would you like to save changes made to theory times file?",
-                    "Save Changes",
-                    MessageBoxButtons.YesNo,
-                    MessageBoxIcon.Question
-                );
-
-                shouldSave = result == DialogResult.Yes;
-            }
+                if (HasChanged)
+                {
+                    var result = MessageBox.Show(
+                        "Would you like to save changes made to the theory times file?",
+                        "Save Changes",
+                        MessageBoxButtons.YesNo,
+                        MessageBoxIcon.Question
+                    );
 
-            if (fileLock != null)
-            {
-                if (shouldSave)
+                    shouldSave = result == DialogResult.Yes;
+                }
+
+                if (fileLock != null)
+                {
+                    if (shouldSave)
+                        _writeTheoryTimesFilePath();
+                    else
+                        _loadTheoryTimesFilePath(TheoryTimesFilePath);
+                }
+                else if (shouldSave && !string.IsNullOrEmpty(TheoryTimesFilePath) && !File.Exists(TheoryTimesFilePath))
+                {
                     _writeTheoryTimesFilePath();
-                else
-                    _loadTheoryTimesFilePath(TheoryTimesFilePath);
-            }
+                }
+            
+            _addTheoryTimesFileWatcher();
+            _unlockFile();
+            LoadedFileContents = null;
```

Comment:
idea: I get that this forces the UI to reload but I think this might have bad side effects because it is used to know if this view needs/can full reload the form, and if it should rewrite the file on disk so that could cause data loss.

if this was done so that when the form is loaded, i think line 327 is the line to update where the view is reset on load, but it currently only reloads the comparison section. Maybe this place should reload the other fields as well?